model-optimizer/extensions/front/tf/ObjectDetectionAPI.py

```diff
@@ -512,19 +515,104 @@ def update_parameter_shape(graph: Graph, match: [SubgraphMatch, None]):
 
 
-class ObjectDetectionAPIPreprocessorReplacement(FrontReplacementFromConfigFileSubGraph):
+class ObjectDetectionAPITransformationsStart(FrontReplacementPattern):
     """
-    The class replaces the "Preprocessor" block resizing input image and applying mean/scale values. Only nodes related
-    to applying mean/scaling values are kept.
+    This is a anchor transformation which is used to distinguish TF OD API models related transformations.
     """
-    replacement_id = 'ObjectDetectionAPIPreprocessorReplacement'
+    enabled = True
+
+    def run_after(self):
+        return [CropAndResizeReplacement, FakeQuantWithMinMaxVarsToQuantize]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        pass
+
+
+class ObjectDetectionAPITransformationsFinish(FrontReplacementPattern):
+    """
+    This is a anchor transformation which is used to distinguish TF OD API models related transformations.
+    """
+    enabled = True
+    # cleanup the graph after applying of TF OD API transformations to remove a lot of unconnected nodes to avoid issues
+    # with shape inference
+    force_clean_up = True
 
     def run_before(self):
         # PadTFToPad inserts Transpose ops for Pad ops inside the sub-graph corresponding to DetectionOutput.
         # But the inputs corresponding to padding values is re-used as inputs for newly created Pad node. This input
         # is removed during removing nodes from the DO sub-graph so the first input to Transpose is missing which
         # results in TransposeOrderNormalizer transformation failure.
-        return [Pack, TransposeOrderNormalizer, PadTFToPad]
+        return [Pack, TransposeOrderNormalizer, PadTFToPad, SqueezeAxis, StandaloneConstEraser, TFSliceToSliceReplacer,
+                KerasRNNOutputConcatenation, KerasRNNInputSlicing]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        pass
+
+
+def get_specific_ops_with_const_inputs(first_node: Node, allowed_ops: list, forward: bool = True):
+    """
+    Returns the list with information about consecutive nodes of operation from "allowed_ops".
+
+    :param first_node: The first node (not included) to start looking for nodes from the "allowed_ops" list
+    :param allowed_ops: list of allowed operations
+    :param forward: flag specifying direction of search
+    :return: list of triplets (Node, const_port_index, const_value)
+    """
+    node = first_node.out_port(0).get_destination().node if forward else first_node.in_port(0).get_source().node
+    result = []  # (Node, port # with constant input, value)
+    while node.soft_get('op') in allowed_ops:
+        for port in (0, 1):
+            if node.in_port(port).get_source().node.has_valid('value'):  # this is a constant input to the node
+                result.append((node, port, node.in_port(port).get_source().node.value.copy()))
+                node = node.out_port(0).get_destination().node if forward else node.in_port(1 - port).get_source().node
+                break
```

Comment:
I think we need to assert to check that node from `allowed_ops` has two input ports for sure, otherwise, passed `allowed_ops` is incorrect. Without this check it can lead to crash in case of incorrect use of this function.

Reply:
Assert has been added.